lua/entities/gmod_wire_expression2/core/bone.lua

```
@@ -110,18 +111,13 @@ end
 
 --- Returns the entity <this> belongs to
 e2function entity bone:entity()
-	return isValidBone(this)
+	return isValidBone(this) or NULL
```

Should also throw an error, i think, although i don’t know for sure if there can be bones without an entity

---

Although I think there can't be a PhysObj without an Entity, so it's worth adding

---

Ok, just made it return NULL in `E2Lib.IsValidBone`
```lua
local function isValidBone(b)
	if type(b) ~= "PhysObj" or not IsValid(b) then return nil, 0 end
	local ent = b:GetEntity()
	if not ent:IsValid() then
		return NULL -- changed here
	end
	return ent
end
```